pkg/services/authn/clients/oauth.go

```diff
@@ -199,6 +199,10 @@ func (c *OAuth) Authenticate(ctx context.Context, r *authn.Request) (*authn.Iden
 	}, nil
 }
 
+func (c *OAuth) IsEnabled() bool {
+	return c.socialService.GetOAuthInfoProvider(c.providerName).Enabled
```

Comment:
`c.socialService.GetOAuthInfoProvider(c.providerName)`can return nil. Probably won't ever return nil, but it would be safer to check the result of `GetOAuthInfoProvider`.

Reply:
I will replace this with `IsClientEnabled` which checks if the client exists

Reply:
Replaced as suggested. Good catch